mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -74,7 +74,7 @@ public List<Metric> getMetricHistory(String runId, String key) {
 
     GetMetricHistory.Response response = mapper
             .toGetMetricHistoryResponse(httpCaller.get(builder.toString()));
-    List<Metric> metrics = response.getMetricsList();
+    List<Metric> metrics = new ArrayList<>(response.getMetricsList());
```

Comment:
The response of `response.getMetricsList()` does not support `addAll`, so it seems pagination didn't work for a long time in Java.